datafusion/core/src/physical_plan/file_format/parquet.rs

```diff
@@ -572,6 +575,15 @@ impl FileOpener for ParquetOpener {
     }
 }
 
+fn should_enable_page_index(
+    enable_page_index: bool,
+    page_pruning_predicate: &Option<Arc<PagePruningPredicate>>,
+) -> bool {
+    enable_page_index
+        && page_pruning_predicate.is_some()
+        && page_pruning_predicate.as_ref().unwrap().filter_number() > 0
```

Comment:
I think we should avoid the `unwrap()` here 

Perhaps something like

```suggestion
        && page_pruning_predicate.as_ref().map(|p| p.filter_number() > 0).unwrap_or(false)
```